google/cloud/storage/async_writer_connection.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_ASYNC_WRITER_CONNECTION_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_ASYNC_WRITER_CONNECTION_H
+
+#include "google/cloud/storage/async_object_requests.h"
+#include "google/cloud/storage/object_metadata.h"
+#include "google/cloud/future.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include "absl/types/variant.h"
+#include <cstdint>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage_experimental {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * An interface to asynchronously perform resumable uploads.
+ *
+ * The client library uses the
+ * `google.storage.v2.StorageService.BidiWriteObject` RPC to perform
+ * asynchronous resumable uploads to Google Cloud Storage. As the name implies,
+ * this is a bi-directional RPC. The `Write()` messages upload the object data.
+ * The `Read()` messages include information about how much of the uploaded data
+ * has been persisted. Only `Write()` messages that explicitly request a 'flush'
```

Comment:
"Flush" doesn't appear to be an element of the interface vocabulary.

Reply:
Rewrote this section to make it clearer we are speaking about the underlying protocol.